lib/src/components/NxFileUpload/SelectedFile.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React, { forwardRef } from 'react';
+import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+import prettyBytes from 'pretty-bytes';
+
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import NxOverflowTooltip from '../NxTooltip/NxOverflowTooltip';
+import NxTooltip from '../NxTooltip/NxTooltip';
+import { SelectedFileProps } from './types';
+
+export { SelectedFileProps };
+
+const formatSize = (size: number) => prettyBytes(size, { minimumFractionDigits: 1, maximumFractionDigits: 1 });
+
+const SelectedFile = forwardRef<HTMLSpanElement, SelectedFileProps>(function
+SelectedFile({ file, onDismiss, descriptionId }: SelectedFileProps, ref) {
```

Comment:
This type does not need to be re-specified since it's already in the `<>`
```suggestion
SelectedFile({ file, onDismiss, descriptionId }, ref) {
```

Reply:
> This type does not need to be re-specified since it's already in the `<>`

So multiple changes here [6d0f9e1](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/6d0f9e118467a162f6dbeca7f56fec9d3ee044c0) to remove the second `SelectedFileProps` assignment. When I removed it initially, it threw an error `'missing props validation'`. So added propTypes for `SelectedFileProps` which caused a few small adjustments with `descriptionId`.

Reply:
> When I removed it initially, it threw an error 'missing props validation'.

We don't care about PropTypes on internal components.  I'd rather you tell Eslint to ignore that.

Reply:
> We don't care about PropTypes on internal components. I'd rather you tell Eslint to ignore that.

Sounds good, I added `eslint-disable react/prop-types` here: [fc52bcf](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/fc52bcf7aca274a26ac65331ff54117ff3b8baeb)